### ISSUE TITLE
Add a character-based RNN example (2017)

### ISSUE THREAD

**Issue by kimmanlui** [fchollet/keras#7598]
Hi 
Previously @jonilaserson provided sample code at  https://github.com/fchollet/keras/issues/19

Here is the code excerpt which output is selected:
```
def mnrnd(probs):
    rnd = numpy.random.random()
    for i in xrange(len(probs)):
        rnd -= probs[i]
        if rnd <= 0:
            return i
    return i
```
But some people writes below (see https://github.com/mineshmathew/char_rnn_karpathy_keras/blob/master/char_rnn_of_karpathy_keras.ipynb)

```
    preds = model.predict(x, verbose=0)[0]
    #print (np.argmax(preds[7]))
    next_index=np.argmax(preds[len(seed_string)-1])
```
Which one is better? 
Thanks

**Comment by samre12** [comment]
The softmax output from the network can be considered as probability distribution over the characters in the vocabulary. So, there can be several ways through which you can predict the next character in the sequence. One of the more commonly used way in practice is to select the most probable character in the distribution, which is given by 
```python
preds = model.predict(x, verbose=0)[0]
    #print (np.argmax(preds[7]))
    next_index=np.argmax(preds[len(seed_string)-1])
```
where as the other way can be to sample a character from the probability distribution (again the character with the maximum softmax score will be the most probable of getting sampled but it is not the case always). This can be done using 
```python
def mnrnd(probs):
    rnd = numpy.random.random()
    for i in xrange(len(probs)):
        rnd -= probs[i]
        if rnd <= 0:
            return i
    return i
```

**Comment by kimmanlui** [comment]
thank you
